lib/safe-pg-migrations/configuration.rb

```diff
@@ -26,29 +27,44 @@ def initialize
       self.backfill_pause = 0.5.second
       self.retry_delay = 1.minute
       self.max_tries = 5
+      self.max_lock_timeout_for_retry = 1.second
       self.sensitive_logger = nil
+      self.increase_lock_timeout_on_retry = false
     end
 
     def lock_timeout=(value)
       raise 'Setting lock timeout to 0 disables the lock timeout and is dangerous' if value == 0.seconds
 
-      unless value.nil? || value < safe_timeout
-        raise ArgumentError, "Lock timeout (#{value}) cannot be greater than safe timeout (#{safe_timeout})"
+      unless value.nil? || (value < safe_timeout && value <= max_lock_timeout_for_retry)
+        raise ArgumentError, "Lock timeout (#{value}) cannot be greater than the safe timeout (#{safe_timeout}) or the
+                              max lock timeout for retry (#{max_lock_timeout_for_retry})"
       end
 
       @lock_timeout = value
     end
 
     def safe_timeout=(value)
-      raise 'Setting safe timeout to 0 disables the safe timeout and is dangerous' unless value
+      unless value && value > 0.seconds
```

Comment:
if value is nil, `value > 0.seconds` will fail

Reply:
If the value is `nil` then `value` is `falsy` then we never go after the &`&`

Reply:
```
irb(main):025> value=nil
=> nil
irb(main):026>
irb(main):027* unless value && value > 0.seconds
irb(main):028*   puts "oups"
irb(main):029> end
```

Reply:
i was probably tired. sorry. :)